### PR TITLE
Trus 3809 - Add Non-Taxable Confirmation Pages for Individual and Agent to Registration

### PR DESCRIPTION
**Pull request by JoshuaH01**

### CODE REVIEW

File: app/controllers/register/ConfirmationController.scala [hunk 1]
```diff
@@ -41,8 +42,10 @@ class ConfirmationController @Inject()(
                                         getData: DraftIdRetrievalActionProvider,
                                         requireData: RegistrationDataRequiredAction,
                                         val controllerComponents: MessagesControllerComponents,
-                                        newIndividualView: newTrust.IndividualView,
-                                        newAgentView: newTrust.AgentView,
+                                        newTaxableIndividualView: newTrust.taxable.IndividualView,
+                                        newTaxableAgentView: newTrust.taxable.AgentView,
+                                        newIndividualView: IndividualView,
+                                        newAgentView: AgentView,
```
Comment (adamconder) [inline app/controllers/register/ConfirmationController.scala:48]: @JoshuaH01  What is the difference between these views? I cannot tell from the names of the values or their imports.

File: app/controllers/register/ConfirmationController.scala [hunk 2]
```diff
@@ -64,16 +67,23 @@ class ConfirmationController @Inject()(
...
+        Future.successful(Ok(newTaxableAgentView(draftId, trn, name)))
+      case (Some(false), true) =>
+        Future.successful(Ok(newTaxableIndividualView(draftId, trn, name)))
+      case (Some(false), false) if isAgent =>
         Future.successful(Ok(newAgentView(draftId, trn, name)))
-      case Some(false) =>
+      case (Some(false), false) =>
         Future.successful(Ok(newIndividualView(draftId, trn, name)))
-      case None =>
+      case (None, _) =>
```
Comment (adamconder) [inline app/controllers/register/ConfirmationController.scala:86]: @JoshuaH01 I guess these are the non-taxable views? Can you name the package or the files to refer to that please.

File: app/models/core/UserAnswers.scala [hunk 3]
```diff
@@ -49,7 +49,8 @@ final case class UserAnswers(
                               data: JsObject = Json.obj(),
                               progress : RegistrationStatus = NotStarted,
                               createdAt : LocalDateTime = LocalDateTime.now,
-                              internalAuthId :String
+                              internalAuthId :String,
+                              isTaxable: Boolean = false
```
Comment (adamconder) [inline app/models/core/UserAnswers.scala:53]: We should probably be default taxable to true?

If this is deployed and a user loads their answers from mongo they will become non-taxable which will break their journey and result them in getting the wrong confirmation page.

File: app/models/core/UserAnswers.scala [hunk 4]
```diff
@@ -109,7 +110,8 @@ object UserAnswers {
       (__ \ "data").read[JsObject] and
       (__ \ "progress").read[RegistrationStatus] and
       (__ \ "createdAt").read(MongoDateTimeFormats.localDateTimeRead) and
-      (__ \ "internalId").read[String]
+      (__ \ "internalId").read[String] and
+      (__ \ "isTaxable").readWithDefault[Boolean](false)
```
Comment (adamconder) [inline app/models/core/UserAnswers.scala:114]: Same for here should be true.

File: app/views/register/confirmation/newTrust/AgentView.scala.html [hunk 5]
```diff
@@ -33,27 +33,37 @@
...
+        <p>@messages("confirmation.nonTaxable.agent.p2")</p>
+        <p>@messages("confirmation.nonTaxable.agent.p3") <a href="@config.helplineUrl" target="_blank">@messages("confirmation.nonTaxable.agent.link1")</a>.</p>
+
+    <h2>@messages("confirmation.nonTaxable.agent.subheading2")</h2>
+        <p>@messages("confirmation.nonTaxable.agent.p4")</p>
+        <p>@messages("confirmation.nonTaxable.agent.p5", leadTrusteeName)</p>
+            <li class="bold">
+                <p>@messages("confirmation.nonTaxable.agent.b1")</p>
+                <p>@messages("confirmation.nonTaxable.agent.b2")</p>
+            </li>
```
Comment (adamconder) [inline app/views/register/confirmation/newTrust/AgentView.scala.html:49]: This is not valid markup. Either they need to be in a `<ul>` or `<ol>`?

File: app/views/register/confirmation/newTrust/AgentView.scala.html [hunk 6]
```diff
@@ -33,27 +33,37 @@
...
+        <p>@messages("confirmation.nonTaxable.agent.p8", leadTrusteeName)</p>
+        <p>@messages("confirmation.nonTaxable.agent.p9")</p>
+        <p>@messages("confirmation.nonTaxable.agent.p10") <a href="config.ggSignInUrl" target="_blank">@messages("confirmation.nonTaxable.agent.link2")</a></p>
+        <p>@messages("confirmation.nonTaxable.agent.p11")</p>
+
+    <h2>@messages("confirmation.nonTaxable.agent.subheading4")</h2>
+        <p>@messages("confirmation.nonTaxable.agent.p12", leadTrusteeName)</p>
+        <p>@messages("confirmation.nonTaxable.agent.p13", leadTrusteeName)</p>
+
+    <h2>@messages("confirmation.nonTaxable.agent.subheading5")</h2>
```
Comment (adamconder) [inline app/views/register/confirmation/newTrust/AgentView.scala.html:63]: Are we confident the heading levels are correct from a markup/accessibility perspective? Are they are as equally important as each other?

File: app/views/register/confirmation/newTrust/AgentView.scala.html [hunk 7]
```diff
@@ -33,27 +33,37 @@
...
+                <p>@messages("confirmation.nonTaxable.agent.b1")</p>
+                <p>@messages("confirmation.nonTaxable.agent.b2")</p>
+            </li>
+        <p>@messages("confirmation.nonTaxable.agent.p6")</p>
+        <p>@messages("confirmation.nonTaxable.agent.p7")</p>
+
+    <h2>@messages("confirmation.nonTaxable.agent.subheading3")</h2>
+        <p>@messages("confirmation.nonTaxable.agent.p8", leadTrusteeName)</p>
+        <p>@messages("confirmation.nonTaxable.agent.p9")</p>
+        <p>@messages("confirmation.nonTaxable.agent.p10") <a href="config.ggSignInUrl" target="_blank">@messages("confirmation.nonTaxable.agent.link2")</a></p>
```
Comment (adamconder) [inline app/views/register/confirmation/newTrust/AgentView.scala.html:56]: This `config.ggSignInUrl` will not render the link correct, have you tested this?

Comment (adamconder) [reply app/views/register/confirmation/newTrust/AgentView.scala.html:56]: It needs the `@` at the start to interpret the scala code.

File: app/views/register/confirmation/newTrust/taxable/AgentView.scala.html [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+    <p>@messages("confirmation.new.agent.p2")</p>
+    <p>@messages("confirmation.new.agent.p3")</p>
+
+    <ol class="govuk-list--number">
+        <li class="bold">@messages("confirmation.new.agent.bullet1")
+            <p>@messages("confirmation.new.agent.bullet1.p1")</p>
+            <p>@messages("confirmation.new.agent.bullet1.p2") <a href="@config.ggSignInUrl" target="_blank">@messages("confirmation.new.agent.bullet1.p2.a")</a>.</p>
+            <p>@messages("confirmation.new.agent.bullet1.p3")</p>
+            <p>@messages("confirmation.new.agent.bullet1.p4")</p>
+        </li>
```
Comment (adamconder) [inline app/views/register/confirmation/newTrust/taxable/AgentView.scala.html:53]: Have you validated this markup? We should not be having <p> inside <li>?

File: conf/messages.en [hunk 9]
```diff
@@ -448,6 +449,41 @@ confirmation.existing.agent.bullet2.p3.a = contact the trusts helpline (opens in
...
 
+confirmation.nonTaxable.individual.subheading1 = What happens next
+confirmation.nonTaxable.individual.p1 = The Unique Reference Number for the trust has been linked to this Government Gateway account. You can return to this service using the same Government Gateway account used to register the trust, to update the trust details.
+confirmation.nonTaxable.individual.subheading2 = Managing the trust
+confirmation.nonTaxable.individual.p2 = This trust does not need to make an annual declaration, or submit an annual tax return. However, you need to keep the details of the trust up to date. You also need to let us know if the trust becomes liable for tax.
+confirmation.nonTaxable.individual.p3 = If you need any help with the service
+confirmation.nonTaxable.individual.link1 = contact the trusts helpline (opens in a new window or tab)
+
+confirmation.nonTaxable.agent.subheading1 = What happens next
+confirmation.nonTaxable.agent.p1 = We will send the trust's Unique Reference Number (URN) to {0} by post.
```
Comment (adamconder) [inline conf/messages.en:460]: Incorrect apostrophe used here and won't render on the view.

**Review by adamconder — CHANGES_REQUESTED** [review]

File: app/views/register/confirmation/newTrust/AgentView.scala.html (hunk 6 above)
Comment (JoshuaH01) [reply app/views/register/confirmation/newTrust/AgentView.scala.html:63]: Confirmed by scott the two subheadings relating to claiming and auth should be h3

**Review by adamconder — APPROVED** [review]

**Pull request merged**